src/index.ts

```diff
@@ -219,63 +77,112 @@ export = function plugin(
       }
     },
 
-    transform(code, filePath) {
+    async transform(code, filePath) {
```

Comment:
Good job @marcalexiei - One thing - I noticed you're using `async` here - Since we're (currently) supporting node versions ">= 10" I'm not sure we can just assume that all clients have support for async/await - Seems we may have to release a new (major) version of the plugin that makes this change.  Thoughts?

Reply:
Also, when I say "we're supporting..." I mean by way of our 'package.json' file.

Reply:
From MDN:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#browser_compatibility

Async functions are supported from Node version 7.6.0. so this shouldn't be a breaking change.
(I haven't tested it out 😅).

Anyway if we are releasing this as a breaking change I would consider to make `api: modern`  the default value.

----

Also I would like to do another change before the next release: 
I'm trying to use `processor` to generate CSS modules:
```js
    sass({
      processor: async (styles, filePath) => {
        let scopedClassNames = {};
        const postcssProcessResult = await postcss([
          postcssModules({
            getJSON: (_, json) => {
              if (json) scopedClassNames = json;
            },
          }),
        ]).process(styles, {
          from: filePath,
        });

        return { ...scopedClassNames, css: postcssProcessResult.css };
      },
```

Right now the plugin always returns the css string as `export default` and I haven't found a clean solution to change it that using `processor` options. 
Have you implemented something similar that I can use or do you know another way?

My possible solution is to allow  `processor` output a new property (e.g.: `cssModuleMap`) that, if present will be used as `default` export. What do you think?

If you want to discuss this further I will open a separate issue since it is out of scope in this PR 😁.